File: SecondMaxElementLinkedList/Find.go
package main
import "fmt"

type Node struct {
	data int;
	next * Node;
}

type NODE = Node;
type PNODE = * Node;
type PPNODE = ** Node;

func InsertFirst(Head PPNODE,value int) {
	var newn PNODE = new(NODE);
	newn.data = value;
	newn.next = nil;

	if(*Head == nil) {
		*Head = newn;
	} else {
		newn.next = *Head;
		*Head = newn;
	}
}

func InsertLast(Head PPNODE,value int) {
	var newn PNODE = new(NODE);
	newn.data = value;
	newn.next = nil;
	var temp PNODE = *Head;

	if(*Head == nil) {
		*Head = newn;
	} else {
		for(temp.next != nil) {
			temp = temp.next;
		}
		temp.next = newn;
	}
}

func Display(Head PNODE) {
	for(Head != nil) {
		fmt.Printf("|%d|->",Head.data);
		Head = Head.next;
	}
}

func Count(Head PNODE) int {
	var icnt int = 0;

	for(Head != nil) {
		icnt++;
		Head = Head.next;
	}

	return icnt;
}

func SecondMax(Head PNODE) int {
	var tempMax int = 0;
	var iSecMax int = 0;
	var temp PNODE = Head;

	tempMax = temp.data;
	for(temp != nil) {
		if(temp.data > tempMax) {
			tempMax = temp.data;
		}
		temp = temp.next;
	}

	iSecMax = Head.data;
	for(Head != nil) {
		if((Head.data > iSecMax)&&(Head.data < tempMax)) {
			iSecMax = Head.data;
		}
		Head = Head.next;
	}

	return iSecMax;
}
func main() {
	var First PNODE = nil;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");
	InsertFirst(&First,51);
	InsertFirst(&First,11);
	InsertFirst(&First,10);
	InsertLast(&First,101);
	InsertLast(&First,150);
	InsertLast(&First,120);

	fmt.Print("Elements of Linked List are : ");
	Display(First);

	iret = Count(First);
	fmt.Println("\nCount of Elements From Linked List is :",iret);

	iret = SecondMax(First);
	fmt.Println("Second Max Element From Linked List is :",iret);
}

File: SecondMaxElementLinkedList/Find.c
#include<stdio.h>
#include<stdlib.h>

struct Node 
{
    int data;
    struct Node * next;
};

typedef struct Node NODE;
typedef struct Node * PNODE;
typedef struct Node ** PPNODE;

void InsertFirst(PPNODE Head,int value)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = value;
    newn->next = NULL;

    if(*Head == NULL)
    {
        *Head = newn;
    }
    else
    {
        newn->next = *Head;
        *Head = newn;
    }
}

void InsertLast(PPNODE Head,int value)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = value;
    newn->next = NULL;
    PNODE temp = *Head;

    if(*Head == NULL)
    {
        *Head = newn;
    }
    else 
    {
        while(temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }
}

void Display(PNODE Head)
{
    while(Head != NULL)
    {
        printf("|%d|->",Head->data);
        Head = Head->next;
    }
}

int Count(PNODE Head)
{
    int icnt = 0;
    while(Head != NULL)
    {
        icnt++;
        Head = Head->next;
    }
    return icnt;
}

int SecondMax(PNODE Head)
{
    int iSecMax = 0;
    int tempMax = 0;
    PNODE temp = Head;

    tempMax = temp->data;
    while(temp != NULL)
    {
        if(temp->data > tempMax)
        {
            tempMax = temp->data;
        }
        temp = temp->next;
    }

    iSecMax = Head->data;
    while(Head != NULL)
    {
        if((Head->data > iSecMax)&&(Head->data < tempMax))
        {
            iSecMax = Head->data;
        }
        Head = Head->next;
    }

    return iSecMax;
}

int main()
{
    PNODE First = NULL;
    int iret = 0;

    printf("Jay Ganesh...\n");

    InsertFirst(&First,51);
    InsertFirst(&First,11);
    InsertFirst(&First,10);
    InsertLast(&First,160);
    InsertLast(&First,150);
    InsertLast(&First,101);
    InsertLast(&First,120);
    
    printf("Elements From Linked list is : \n");
    Display(First);
    
    iret = Count(First);
    printf("\nCount Of Elements From Linked list is : %d\n",iret);

    iret = SecondMax(First);
    printf("Second Max Element From Linked List is : %d\n",iret);

    return 0;
}